source/main.bmx

```diff
@@ -5326,19 +5326,24 @@ endrem
 
 				toast.SetCaption( caption )
 				If i = room.owner
+					Local player:TPlayer = GetPlayer(room.owner)
 					If room.GetNameRaw() = "news"
 						player.getProgrammePlan().RemoveNewsBySlot(0)
 						player.getProgrammePlan().RemoveNewsBySlot(1)
 						player.getProgrammePlan().RemoveNewsBySlot(2)
 					EndIf
 					Local cost:Int = triggerEvent.GetData().GetInt("renovationBaseCost")
 					If cost > 0
-						Local player:TPlayer = GetPlayer(room.owner)
-						If player
-							cost:* player.GetAudienceReachLevel()
-							'TODO localize
-							text:+ "~nRenovierungskosten " + MathHelper.DottedValue(cost)+getLocale("CURRENCY")
-							player.GetFinance().PayMisc(cost)
+						cost= TFunctions.RoundToBeautifulValue(cost * 1.2^ (player.GetAudienceReachLevel()-1))
```

Comment:
Wieso findet das Runden "nur" fuer die Toast-Message statt? Sollten die "cost" nicht schon beim Eintreten des Schadens "gerundet" werden (da dort ja schon der AudienceReachLevel klar ist).

Reply:
Natürlich. Wird repariert

Reply:
Ich habe mir die Stelle nochmal angeschaut. Das Runden erfolgt ja vor dem Bezahlen. Die Berechnung der finalen Kosten habe ich nicht in room.base gemacht, da man dafür dort eine zusätzliche Abhängigkeit für die Ermittlung der Reichweite ergänzen müsste. Das wollte ich vermeiden. Im Event werden also nur die Basiskosten übergeben und hier für den finalen Effekt (Bezahlen) und die Nachricht weiterverarbeitet. Andere Effekte ergeben sich ja auch erst hier (Lizenz zerstört...)

Ist das OK oder sollen die finalen Renovierungskosten im Raum selbst ermittelt werden?